WordPress/src/androidTest/java/org/wordpress/android/InitializationRule.kt

```diff
@@ -0,0 +1,37 @@
+package org.wordpress.android
+
+import androidx.test.platform.app.InstrumentationRegistry
+import dagger.hilt.EntryPoint
+import dagger.hilt.EntryPoints
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+class InitializationRule : TestRule {
```

Comment:
Could this be named TestInitializationRule for better?

Reply:
`InitializationRule` doesn't initialize tests, it initializes `AppInitializer`. It is a `TestRule` that creates `AppInitializer` and run its `init()` function. Hilt is normally doing that in the application class, but in tests, we're doing it manually with `InitializationRule`. Other TestRule names in [BaseTest](https://github.com/wordpress-mobile/WordPress-Android/blob/a5f0ea7b8488f1ca8eb12aaf2d6ec5a51e607fa2/WordPress/src/androidTest/java/org/wordpress/android/support/BaseTest.java) are `HiltAndroidRule`, `ActivityScenarioRule`, `WireMockRule`. These also don't start with "Test". 
If you still think it could be better to name it `TestInitializationRule`, we can do it. wdyt?